src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/create/actions/OnReceivedResponseAction.java

```diff
@@ -79,7 +79,7 @@ protected void perform(
                 stateMachine.removePendingCommand(commandId);
 
                 response.getFailedCommandIds().forEach((uuid, message) ->
-                        stateMachine.saveErrorToHistory(format(FAILED_TO_APPLY_RULE_MESSAGE, actionName), format(
+                        stateMachine.saveHaFlowActionToHistory(format(FAILED_TO_APPLY_RULE_MESSAGE, actionName), format(
```

Comment:
It must be `saveHaFlowErrorHistory`

Reply:
good catch! I am wondering whether it would be better to have different parameters in the method signature? Maybe create some enum with the error types and require it here? Essentially move this constant string to enum